### PR TITLE
[connman] Apply upstream rtnl patches, revert debug disable commit. JB#57933

### PR DESCRIPTION
**Pull request by LaakkonenJussi**
Applied a patch that fixes the alignment of netlink messages. And also couple of others non-related to maintain consistency in rtnl.c.

Reverted old debug disable commit that was not just proper. `DBG()`s are handled via bootup option or D-Bus toggle with dbuslog-tools.

### CODE REVIEW

File: connman/src/detect.c [hunk 1]
```diff
@@ -27,7 +27,13 @@
 
 #include "connman.h"
 
+struct device_details {
+	enum connman_device_type type;
+	int index;
+};
+
 static GSList *device_list = NULL;
+static gint delayed_newlink_ids[MAX_CONNMAN_SERVICE_TYPES] = { 0 };
```
Comment (inzanity) [inline connman/src/detect.c:36]: Can we relay on there only being one new link per service type?

Comment (LaakkonenJussi) [reply connman/src/detect.c:36]: Yes but this is more of an idea level approach still that might this be the cause even. Which is why I just added commit to add delayed only when there is no MAC. Didn't want to overdo this yet.

Comment (LaakkonenJussi) [reply connman/src/detect.c]: Yep, this made things worse when there is no MAC set for a device. Removed.

**Comment by LaakkonenJussi** [comment]
> @inzanity Added few untested commits, which are more of a workaround for the issue when there are systems that do report the netlink message BEFORE there is a MAC set, I've seen this (as a result MAC is 000000000000):
> 
> ```
> connmand[615]: eth0 no DEVTYPE, defaulting to ethernet
> kernel: rk_gmac-dwmac fe300000.ethernet: rk_get_eth_addr: mac address: xx:xx:xx:xx:xx:xx
> kernel: eth0: device MAC address xx:xx:xx:xx:xx:xx
> ```
> 
> And it can happen in boot, not 100% sure but this would avoid any odd races with kernel regarding this. It would be better to fix the kernels I guess but...
> 
> Haven't tested that yet at all. Too late, Friday.

Removed, only the upstream patches remain and reverting of old commit.

**Pull request merged**